include/wlr/types/wlr_xcursor_manager.h

```diff
@@ -58,12 +63,19 @@ struct wlr_xcursor *wlr_xcursor_manager_get_xcursor(
 	struct wlr_xcursor_manager *manager, const char *name, float scale);
 
 /**
- * Set a wlr_cursor's cursor image to the specified cursor name for all scale
- * factors. wlr_cursor will take over from this point and ensure the correct
- * cursor is used on each output, assuming a wlr_output_layout is attached to
- * it.
+ * Retrieves the cached texture of this xcursor image from the
+ * wlr_xcursor_manager.
+ *
+ * The wlr_xcursor_manager must have been created with a wlr_renderer and the
+ * wlr_xcursor_image must have come from a wlr_xcursor returned from
+ * wlr_xcursor_manager_get_xcursor.
+ *
+ * The returned texture is owned by the xcursor manager. You should not save or
+ * destroy it.
+ * The manager makes use of the wlr_xcursor_image.userdata field. You must not
+ * modify this field yourself.
```

Comment:
Maybe we can rejigger things a bit so that this field is in a private struct.

Reply:
I didn't really want to make an entire copy of the `wlr_xcursor_theme` "tree". I figured a userdata field would be the simplest and most elegant solution, and could help people not using `wlr_renderer` when that truly becomes viable.

Reply:
Why does moving this to a private struct necessitate a copy of the `wlr_xcursor_theme` "tree"? Unsure of what "tree" means here, either.